src/app/user/user-details/user-details.component.html

```diff
@@ -22,37 +22,37 @@ <h1>User Details</h1>
         <div class="details-item">
           <div class="title-container">
             <mat-icon>title</mat-icon>
-            <h2>First Name</h2>
+            <h2>{{ "USER-DETAILS.FNAME" | translate }}</h2>
           </div>
           <p>{{ user.firstName }}</p>
         </div>
         <div class="details-item">
           <div class="title-container">
             <mat-icon>title</mat-icon>
-            <h2>Last Name</h2>
+            <h2>{{ "USER-DETAILS.LNAME" | translate }}</h2>
           </div>
           <p>{{ user.lastName }}</p>
         </div>
         <div class="details-item">
           <div class="title-container">
             <mat-icon>badge</mat-icon>
-            <h2>Username</h2>
+            <h2>{{ "USER-DETAILS.USERNAME" | translate }}</h2>
           </div>
           <p>{{ user.username }}</p>
         </div>
         <div class="details-item">
           <div class="title-container">
             <mat-icon>lock</mat-icon>
-            <h2>Roles</h2>
+            <h2>{{ "USER-DETAILS.ROLES" | translate }}</h2>
           </div>
           @if (this.editable === false) {
             @for (role of user.roles; track $index) {
-              <p>{{ role }}</p>
+              <p>{{ role | translate }}</p>
             }
           } @else {
             <mat-select formControlName="roles" multiple>
-              <mat-option [disabled]="true" value="User">User</mat-option>
-              <mat-option value="Manager">Manager</mat-option>
+              <mat-option [disabled]="true" value="User">{{ "User" | translate }}</mat-option>
+              <mat-option value="Manager">{{ "Manager" | translate }}</mat-option>
```

Comment:
keep the translations the same structure, as it seems now you have an object lets say NAV and under that you have different navigation tabs, same for roles can be created ROLES -> MANAGER, USER, ADMIN